### DISCUSSION TITLE
Font size change in screen resize

### DISCUSSION THREAD

**Discussion by sadrajafari** [scenerystack/community#26] [Q&A]
Hey I am just wondering why the font size does not update in different screen sizes.

**Comment by jessegreenberg** [comment]
Setting the font size with a number will set the value in pixels, an absolute unit that does not change with screen size.

```js
      font: new phet.scenery.Font( { size: 20 } )
```

is similar to "20px sans-serif" CSS font. See https://github.com/phetsims/scenery/blob/f2c577900fdc87799e206b75137301dedd5ebd6e/js/util/Font.ts for other font options.

You can update the font size with [Text.setFontSize](https://phetsims.github.io/scenery/doc/#text-fontSize). You can also scale your Text (or any scenery Node) with `setScaleMagnitude` in your layout function.

PhET's simulations scale to screen size by setting a layout matrix on the scene like this: https://github.com/phetsims/joist/blob/16a150d3fae55a7c81728329dc5ed0a528f42da1/js/ScreenView.ts#L291

For example:

<details>


```js

// ... other scenery setup

// A scene that everything will be added to - the root Node cannot be transformed
const scene = new phet.scenery.Node();
rootNode.addChild( scene );

// add circles and tooltips to the scene Node
// ...

const layoutBounds = new phet.dot.Bounds2( 0, 0, 800, 600 );

// Manage layout
const resizeListener = () => {
  const viewBounds = new phet.dot.Bounds2( 0, 0, window.innerWidth, window.innerHeight );

  // now layout is relative to the layoutBounds
  circleA.leftCenter = layoutBounds.leftCenter.plusXY( 50, 0 );
  circleB.rightCenter = layoutBounds.rightCenter.minusXY( 50, 0 );
  tooltipA.centerBottom = circleA.centerTop;
  tooltipB.centerBottom = circleB.centerTop;

  // Compute a layout matrix that will fit and center everything in the screen
  scene.matrix = phet.joist.ScreenView.getLayoutMatrix( layoutBounds, viewBounds );
};
```

</details>
